app/modules/intelligence/agents/chat_agents/code_gen_chat_agent.py

```
@@ -16,20 +16,27 @@
-        self.llm = llm
+        self._llm = None
+        self._llm_provider = ProviderService(db)
         self.history_manager = ChatHistoryService(db)
         self.prompt_service = PromptService(db)
         self.agents_service = AgentsService(db)
         self.chain = None
         self.db = db
 
+    async def _get_llm(self):
+        if self._llm is None:
+            self._llm = await self._llm_provider.get_small_llm(agent_type=AgentType.LANGCHAIN)
```

Ruff (0.8.2)

Undefined name `AgentType`

(F821)